hmcl: fail to launch old minecraft versions with openjdk8

### Nixpkgs version

- Stable (25.11)

### Describe the bug

I have both `hmcl` and `openjdk8` installed with `home-manager`, and old versions of minecraft (requires java8) will crash on launch.

`steam-run hmcl` also crashed, and here's my workaround:

```nix home.nix
    # fhs environment
    (let base = pkgs.appimageTools.defaultFhsEnvArgs; in
      pkgs.buildFHSEnv (base // {
      name = "fhs";
      targetPkgs = pkgs:
        (base.targetPkgs pkgs) ++ (with pkgs; [
          pkg-config
          ncurses
        ]
      );
      profile = "export FHS=1";
      runScript = "${pkgs.writeShellScript "fhs-wrapper" ''
          if [ $# -gt 0 ]; then
            exec "$@"
          else
            exec zsh
          fi
      ''}";
      extraOutputsToInstall = ["dev"];
    }))
  ];
```

`fhs hmcl` works fine.

### Steps to reproduce

1. install `hmcl` and `openjdk8`
2. download minecraft 1.8.9 (or any version that requires java8)
3. launch it

### Expected behaviour

old versions of minecraft work fine

### Screenshots

_No response_

### Relevant log output

```console
---- Minecraft Crash Report ----
// My bad.

Time: 9/1/26 10:11 AM
Description: Initializing game

java.lang.ExceptionInInitializerError
	at ave.ar(SourceFile:560)
	at ave.am(SourceFile:361)
	at ave.a(SourceFile:310)
	at net.minecraft.client.main.Main.main(SourceFile:124)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
	at org.lwjgl.opengl.LinuxDisplay.getAvailableDisplayModes(LinuxDisplay.java:951)
	at org.lwjgl.opengl.LinuxDisplay.init(LinuxDisplay.java:738)
	at org.lwjgl.opengl.Display.<clinit>(Display.java:138)
	... 4 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at ave.ar(SourceFile:560)
	at ave.am(SourceFile:361)

-- Initialization --
Details:
Stacktrace:
	at ave.a(SourceFile:310)
	at net.minecraft.client.main.Main.main(SourceFile:124)

-- System Details --
Details:
	Minecraft Version: 1.8.9
	Operating System: Linux (amd64) version 6.12.63
	Java Version: 1.8.0_472, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 85241648 bytes (81 MB) / 167772160 bytes (160 MB) up to 3523215360 bytes (3360 MB)
	JVM Flags: 17 total; -Xmx3332m -XX:+UnlockExperimentalVMOptions -XX:+UnlockDiagnosticVMOptions -XX:+UseG1GC -XX:G1MixedGCCountTarget=5 -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32m -XX:-OmitStackTraceInFastThrow -XX:MaxInlineLevel=15 -XX:-DontCompileHugeMethods -XX:MaxNodeLimit=240000 -XX:NodeLimitFudgeFactor=8000 -XX:TieredCompileTaskTimeout=10000 -XX:ReservedCodeCacheSize=400M -XX:NmethodSweepActivity=1
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	Launched Version: 1.8.9
	LWJGL: 2.9.4
	OpenGL: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
	GL Caps: 
	Using VBOs: No
	Is Modded: Probably not. Jar signature remains and client brand is untouched.
	Type: Client (map_client.txt)
	Resource Packs: file/Fullbright-UB-1.21.zip
	Current Language: ~~ERROR~~ NullPointerException: null
	Profiler Position: N/A (disabled)
	CPU: <unknown>
```

### Additional context

WM: sway (I guess it works fine in X11)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.63, NixOS, 25.11 (Xantusia), 25.11.20260104.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/afk8r2xf4s01dcz833chvkis81ldrziv-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/hm/hmcl/package.nix
+++ b/pkgs/by-name/hm/hmcl/package.nix
@@ -176,7 +176,9 @@ stdenv.mkDerivation (finalAttrs: {
     makeShellWrapper ${hmclJdk}/bin/java $out/bin/hmcl \
       --add-flags "-jar $out/lib/hmcl/hmcl-terracotta-patch.jar" \
       --set LD_LIBRARY_PATH ${lib.makeLibraryPath finalAttrs.runtimeDeps} \
-      --prefix PATH : "${lib.makeBinPath minecraftJdks}" \
+      --prefix PATH : "${
+        lib.makeBinPath (minecraftJdks ++ lib.optional stdenv.hostPlatform.isLinux xorg.xrandr)
+      }" \
       --run 'cd $HOME' \
       ''${gappsWrapperArgs[@]}
   '';

